src/tools/nbench/nbench.cpp

```diff
@@ -269,117 +287,123 @@ OPTIONS
         return 1;
     }
 
-    if (visualize)
+    vector<string> models;
+    if (!directory.empty())
     {
-        shared_ptr<Function> f = deserialize(model);
-        auto model_file_name = ngraph::file_util::get_file_name(model) + std::string(".") +
-                               pass::VisualizeTree::get_file_ext();
-
-        pass::Manager pass_manager;
-        pass_manager.register_pass<pass::VisualizeTree>(model_file_name);
-        pass_manager.run_passes(f);
+        vector<PerfShape> aggregate_perf_data;
+        file_util::iterate_files(directory,
+                                 [&](const string& file, bool is_dir) {
+                                     if (!is_dir)
+                                     {
+                                         models.push_back(file);
+                                     }
+                                 },
+                                 true);
+    }
+    else
+    {
+        // Error case where model is missing already checked above
+        models.push_back(model_arg);
     }
 
-    if (statistics)
+    vector<PerfShape> aggregate_perf_data;
+    for (const string& model : models)
     {
-        shared_ptr<Function> f = deserialize(model);
+        cout << "\n============================================================================"
+                "\n";
+        cout << "---- Processing '" << model << "'\n";
+        cout << "============================================================================\n";
+        if (visualize)
+        {
+            shared_ptr<Function> f = deserialize(model);
```

Comment:
Is it desired behavior for nbench to not catch this exception:

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Error parsing json at node 'ScalarConstantLike_7958'
Aborted
```

Possible alternative: catch and log the throw, move to the next model and report the error at-exit.